test/Table_Tests/src/IO/Fixed_Width_Read_Spec.enso

```diff
@@ -0,0 +1,238 @@
+
+from Standard.Test import all
+import Standard.Test.Test_Environment
+
+from project.Common_Table_Operations.Util import run_default_backend
+
+add_specs suite_builder =
+    data_dir = enso_project.data / "fixed-width"
+    data0 = data_dir / "fixed0.txt"
+    data1 = data_dir / "fixed1.txt"
+    data2 = data_dir / "fixed2.txt"
+    data3 = data_dir / "fixed3.txt"
+
+    suite_builder.group "Read fixed-width files" group_builder->
+        group_builder.specify "Should make a layout from a description" <|
+            descs = []
+                + [Fixed_Width_Column_Description.Start_And_Width "a" 0 4]
+                + [Fixed_Width_Column_Description.Width "b" 1]
+                + [Fixed_Width_Column_Description.Width "c" 3]
+                + [Fixed_Width_Column_Description.Width "d" 3]
+            layout = Fixed_Width_Layout.from_descriptions descs
+            expected = []
+                + [Fixed_Width_Layout_Entry.Value "a" 0 4]
+                + [Fixed_Width_Layout_Entry.Value "b" 4 1]
+                + [Fixed_Width_Layout_Entry.Value "c" 5 3]
+                + [Fixed_Width_Layout_Entry.Value "d" 8 3]
+            layout . get_entries . should_equal expected
+
+        group_builder.specify "Should make a layout from a table (no column names)" <|
+            table = Table.input [['x', [0, Nothing, Nothing, Nothing]], ['y', [4, 1, 3, 3]]]
+            layout = table : Fixed_Width_Layout
+            expected = []
+                + [Fixed_Width_Layout_Entry.Value "Column 1" 0 4]
+                + [Fixed_Width_Layout_Entry.Value "Column 2" 4 1]
+                + [Fixed_Width_Layout_Entry.Value "Column 3" 5 3]
+                + [Fixed_Width_Layout_Entry.Value "Column 4" 8 3]
+            layout . should_be_a Fixed_Width_Layout
+            layout . get_entries . should_equal expected
+
+        group_builder.specify "Should make a layout from a table (column names)" <|
+            table = Table.input [['c', ["a", "b", "c", "d"]], ['x', [0, Nothing, Nothing, Nothing]], ['y', [4, 1, 3, 3]]]
+            layout = table : Fixed_Width_Layout
+            expected = []
+                + [Fixed_Width_Layout_Entry.Value "a" 0 4]
+                + [Fixed_Width_Layout_Entry.Value "b" 4 1]
+                + [Fixed_Width_Layout_Entry.Value "c" 5 3]
+                + [Fixed_Width_Layout_Entry.Value "d" 8 3]
+            layout . should_be_a Fixed_Width_Layout
+            layout . get_entries . should_equal expected
+
+        group_builder.specify "Should reject a format that tries to go to an earlier column" <|
+            descs = []
+                + [Fixed_Width_Column_Description.Width "a" 4]
+                + [Fixed_Width_Column_Description.Start_And_Width "b" 3 1]
+                + [Fixed_Width_Column_Description.Width "c" 3]
+            layout = Fixed_Width_Layout.from_descriptions descs
+            layout.should_fail_with Illegal_Argument
+            layout.catch.message . should_contain "must start at column"
```

Comment:
Nice test. Shall we also have a test for negative column width?

What about column width `== 0`? Should it be allowed? I guess it's not impossible, just will create an empty column which is useless.

Reply:
Oh now I see this is tested below, great.